configs/xlml/pytorch/multipodteam_llama2.py

```diff
@@ -0,0 +1,123 @@
+    logging_dir: str,
+    per_slice_batch_size: int,
+    block_size: int = 2048,
+    dataset_name: str = "wikitext",
+    dataset_config: str = "wikitext-2-raw-v1",
+    num_slices: int = 1,
+    save_strategy: str = "no",
+    chkpt_manager_path: Optional[str] = None,
+    max_steps: Optional[int] = None,
+    xla_execution_time_step: Optional[int] = None,
+):
+  command = (
+      "python3 /tmp/transformers/examples/pytorch/language-modeling/run_clm.py "
+      "--tokenizer_name hf-internal-testing/llama-tokenizer "
+      f"--dataset_name {dataset_name} "
+      f"--dataset_config_name {dataset_config} "
+      # TODO(jonbolin): SPMD requires integration with Accelerate for
+      # distributed dataloading. per_device_train_batch_size currently accepts
+      # the global batch size.
+      f"--per_device_train_batch_size {per_slice_batch_size * num_slices} "
+      "--do_train "
+      "--output_dir /tmp/output "
+      f"--config_name /configs/llama/{config_name}.json "
+      "--remove_unused_columns no "
+      "--report_to tensorboard "
+      f"--logging_dir {logging_dir} "
+      "--spmd_defer_init "
+      "--spmd_2d_sharding 1 "
+      f"--spmd_dcn_parallelism {num_slices} "
+      "--optim adafactor "
+      "--torch_dtype float32 "
+      "--dataloader_drop_last yes "
+      "--spmd_grad_chkpt "
+      f"--block_size {block_size} "
+  )
+  if chkpt_manager_path:
+    command = ' '.join((command, f'--checkpoint_manager_path {chkpt_manager_path}'))
+  if max_steps:
+    command = ' '.join((command, f'--max_steps {max_steps}'))
+  if xla_execution_time_step is not None:
+    command = ' '.join((command, f'--xla_execution_time_step {xla_execution_time_step}'))
+  return command
+
+
+def get_pytorch_llama2_perf_config(
+  tpu_version: str,
+  tpu_cores: int,
+  tpu_zone: str,
+  cluster_name: str,
+  cluster_project_name: str,
+  gcs_bucket_prefix: str,
+  per_slice_batch_size: int = 16,
+  config_name: str = "2B",
+  num_slices: int = 1,
+  time_out_in_min: int = 60,
+):
+  job_gcp_config = gcp_config.GCPConfig(
+    project_name=cluster_project_name,
+    zone=tpu_zone,
+    dataset_name=metric_config.DatasetOption.XLML_DATASET,
+  )
+
+  gcs_bucket_path = os.path.join(gcs_bucket_prefix, f"{num_slices}xv{tpu_version}-{tpu_cores}")
+
+  hf_setup_commands = common.set_up_hugging_face_transformers_llama2_fork()
+  train_command = _get_run_clm_command(config_name=config_name, logging_dir=gcs_bucket_path, per_slice_batch_size=per_slice_batch_size, max_steps=10, xla_execution_time_step=5)
+  # TODO(jonbolin): set_up_cmds isn't used for GKE workloads, put setup in
+  # run_model_cmds
+  run_model_cmds = (*hf_setup_commands, train_command)
+
+  job_test_config = test_config.TpuGkeTest(
+    test_config.Tpu(
+      version=tpu_version,
+      cores=tpu_cores,
+    ),
+    test_name=f"pytorch_spmd_llama2_{config_name}_perf",
+    cluster_name=cluster_name,
+    docker_image=vm_resource.DockerImage.PYTORCH_MULTISLICE_BASELINE.value,
```

Comment:
cc @RissyRan, the image here is being used to launch a container to run `xpk.py workload create` within my test project: http://shortn/_ABDbVB3KIF

I expected this image to be used only for the actual workload on the TPU VMs, so the permissions aren't configured properly to pull it.

Reply:
Hi Jon, thanks for bring this up!

We should have 2 images. [1st image](https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/implementations/utils/xpk.py#L80) is used for pod in env to work around Python3.10 used by xpk, which is not supported in Composer yet. [2nd image](https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/implementations/utils/xpk.py#L70) is used for xpk experiments.

Like you mentioned, we could just update the 1st one to python3.10:slim.